presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformSpatialPredicates.java

```diff
@@ -110,50 +114,117 @@
 
     private final Metadata metadata;
 
-    public TransformSpatialPredicateToJoin(Metadata metadata)
+    public TransformSpatialPredicates(Metadata metadata)
     {
         this.metadata = metadata;
     }
 
-    @Override
-    public Pattern<FilterNode> getPattern()
+    public Set<Rule<?>> rules()
     {
-        return PATTERN;
+        return ImmutableSet.of(
+                new TransformSpatialPredicateToJoin(metadata),
+                new TransformSpatialPredicateToLeftJoin(metadata));
     }
 
-    @Override
-    public boolean isEnabled(Session session)
+    public static final class TransformSpatialPredicateToJoin
```

Comment:
@arhimondr I don't think I can do either of these. I can't use `rules()` because I want to test individual rules and I can't make them package private because test is in a different package. The test cannot be in the same package because it has to be in a different module that can depend on presto-geospatial plugin.